content/courses/advanced-ada/parts/data_types/shared_variable_control.rst

```diff
@@ -469,20 +473,806 @@ these array declarations are equivalent:
+
+          --  unspecified
+          Reserved_14_15 : UInt2  := 16#0#;
+
+          --  vertical line count
+          Vertical_Cnt   : UInt14 := 16#0#;
+
+          --  refresh signalling
+          Refresh_Needed : Boolean_Bit := False;
+
+          --  unspecified
+          Reserved_30    : UInt1  := 16#0#;
+       end record
+         with Size      => 32,
+              Bit_Order => System.Low_Order_First,
+              Volatile,
+              Full_Access_Only;
+
+       for Window_Register use record
+          Horizontal_Cnt at 0 range 0 .. 13;
+          Reserved_14_15 at 0 range 14 .. 15;
+          Vertical_Cnt   at 0 range 16 .. 29;
+          Refresh_Needed at 0 range 30 .. 30;
+          Reserved_30    at 0 range 31 .. 31;
+       end record;
+
+       procedure Show (WR : Window_Register);
+
+    end Registers;
+
+    with Ada.Text_IO; use Ada.Text_IO;
+
+    package body Registers is
+
+       procedure Show (WR : Window_Register) is
+       begin
+          Put_Line ("WR = (Horizontal_Cnt => "
+                    & WR.Horizontal_Cnt'Image
+                    & ",");
+          Put_Line ("      Vertical_Cnt   => "
+                    & WR.Vertical_Cnt'Image
+                    & ",");
+          Put_Line ("      Refresh_Needed => "
+                    & WR.Refresh_Needed'Image
+                    & ")");
+       end Show;
+
+    end Registers;
+
+In this example, we have a 32-bit register (of :ada:`Window_Register` type)
+that contains window information for a display:
+
+.. graphviz::
+
+    digraph foo {
+        "Record_R" [
+            label = "{ position | bits | component } |  { 0 | { { #0 .. 13 | Horizontal_Cnt } | { #14 .. #15 | Reserved_14_15 } | { #16 .. #29 | Vertical_Cnt } | { #30 .. #31 | Reserved_30_31 } } }"
+            shape = "record"
+        ];
+   }
+
+Let's use the :ada:`Window_Register` type from the :ada:`Registers` package in
+a test application:
+
+.. code:: ada run_button project=Courses.Advanced_Ada.Data_Types.Type_Representation.Shared_Variable_Control.Nonatomic_Full_Access_Register
+
+    pragma Ada_2022;
+
+    with Registers;   use Registers;
+
+    procedure Show_Register is
+       WR : Window_Register;
+    begin
+       --  Nonatomic full-access assignments
+       WR.Horizontal_Cnt := 800;
+       WR.Vertical_Cnt   := 600;
+       WR.Refresh_Needed := True;
 
-    :ada:`Full_Access_Only`
-    -----------------------
+       Show (WR);
+    end Show_Register;
 
-    .. admonition:: Relevant topics
+The example contains assignments such as :ada:`WR.Horizontal_Cnt := 800` and
+:ada:`WR.Vertical_Cnt:= 600`. Because :ada:`Window_Register` is a full-access
+type, these assignments are performed for the complete 32-bit register, even
+though we're updating just a single component of the record object. (Note that
+if :ada:`Window_Register` wasn't a *full-access* object, an assignment such as
+:ada:`WR.Horizontal_Cnt := 800` could be performed with a 16-bit operation.)
```

Comment:
Regarding "could be performed with a 16-bit operation": This is in an admonition, but it is central to the concept because it is what the compiler would want to do otherwise, because that is more efficient than manipulating the entire object.

Reply:
Good point! I've just added a paragraph for that topic. Please let me know what you think!

Reply:
Looks good!